### DISCUSSION TITLE
How to initialize cluster in runtime?

### DISCUSSION THREAD

**Discussion by Degot** [microsoft/garnet#561] [Q&A] [answered]
Hello, I have my application with own cluster management.. Each node of my application starts GarnetServer and notifies others... Other nodes execute CLUSTER MEET new-node-ip port.
CLUSTER NODES returns all nodes. How to distribute slots between nodes? How to implement Master-Master replication?

**Comment by vazois** [comment, marked as answer]
You can use commands like cluster addslots|addslotsrange. There are more details on how to use them in our website
https://microsoft.github.io/garnet/docs/commands/cluster
https://microsoft.github.io/garnet/docs/cluster/overview
